pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,46 @@
+package upgrade
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+// ManagedInstances finds all the Jaeger instances for the current operator and upgrades them, if necessary
+func ManagedInstances(c client.Client) error {
+	list := &v1.JaegerList{}
+	opts := &client.ListOptions{}
+	if err := c.List(context.Background(), opts, list); err != nil {
```

Comment:
If multiple operators are deployed with different WATCH_NAMESPACE lists - how do we ensure that the managed instances returned were managed by this operator?

Reply:
We can either use a predicate as part of ListOptions, or do a manual check on the results. IIRC, we can only filter by labels, and I we might be using an annotation indicating which operator is managing the instance.

Reply:
A solution for this would still be required.

Reply:
I'll fix this as part of #567, as I could make use of the new `ManagedBy` field here.